security/aws/rds_unencrypted/AWS_Unencrypted_RDS_Instances.pt

```diff
@@ -0,0 +1,493 @@
+        }
+        // Constructing tags with comma separated to display in detail_template
+        if((tag['tagValue']).length > 0){
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']+'='+tag['tagValue']
+        }else{
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']
+        }
+      }
+
+      var publicly_Accessible = rds['publicly_Accessible'];
+      if(publicly_Accessible === true){
+        publicly_Accessible = 'Yes';
+      }else{
+        publicly_Accessible = 'No';
+      }
+
+      var delete_Protection = rds['delete_Protection'];
+      if(delete_Protection === true){
+        delete_Protection = 'Yes';
+      }else{
+        delete_Protection = 'No';
+      }
+
+      //If the RDS instance tag does not match with entered param_exclude_tags, then check if RDS instance is Unencrypted.
+      if(!(isTagMatched) && (rds['storage_Encrypted'] === false)){
+        content.push({
+          dbInstance_Identifier: rds['dbInstance_Identifier'],
+          region: rds['region'],
+          availability_zone: rds['availability_zone'],
+          sec_availability_zone: rds['sec_availability_zone'],
+          publicly_Accessible: publicly_Accessible,
+          dbName: rds['dbName'],
+          db_instance_status: rds['db_instance_status'],
+          delete_Protection: delete_Protection,
+          tagKeyValue:(tagKeyValue.slice(2)),
+          dbInstanceArn: rds['dbInstanceArn'],
+          dbCluster_Identifier: rds['dbCluster_Identifier']
+        })
+      }
+    }
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_rds_instance_list" do
+  validate $ds_rds_unencrypted_instance_list do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} unencrypted RDS Instances Found in AWS"
+    detail_template <<-EOS
+# List of unencrypted RDS Instances
+| RDS Instance Name | Region | Availability Zone | Sec Availability Zone | Publicly Accessible | DB Name | DB Instance Status | Delete Protection Enabled | TAGs |
+| ----------------- | ------ | ----------------- | --------------------- | ------------------- | ------- | ------------------ | ------------------------- | ---- |
+{{ range data -}}
+| {{.dbInstance_Identifier}} | {{ .region }} | {{ .availability_zone }} |  {{ .sec_availability_zone }} |{{ .publicly_Accessible }} | {{ .dbName }} | {{ .db_instance_status }} | {{.delete_Protection}} | {{.tagKeyValue}} |
+{{ end -}}
+###### Note: Kindly refer 'Enable Delete Action' section in the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/rds_unencrypted) file to perform delete operation.
+  EOS
+    escalate $report_unencrypted_RDS_instances
+    #Uncomment (remove '#') in the below line to enable Delete action.
+    #escalate $delete_unencrypted_RDS_instances_approval
+    check eq(size(data),0)
```

Comment:
If its validate_each, check will be different -- wouldn't we want to check that delete_protection field, i.e check eq(val(item, "delete_protection"), "YES") or something. In the escalation node it says the status has to be available as well so maybe:
```
check logic_or(
  eq(val(item, "delete_protection"), "YES"),
  ne(val(item, "db_instance_status"), "available")
)

```

Reply:
The other problem with doing some of the validation logic with regard to delete_protection in the cloud workflow, this policy will be "triggered" when theres nothing to delete. For example, if we have 1 rds instance thats unencrypted with "delete_protection" on, this policy will trigger and send it to CWF which will skip it over -- it'll stay in a triggered state when there's nothing to do. It'd be better if the policy skipped over that in the validation section, so it can go to an untriggered state.

Reply:
@psschroeter completely agree with you, since the scope of this policy is to report all unencrypted RDS instances irrespective of  'delete protection', 'DB instance status' and with disabled delete action. So I was checking  'delete protection' in the cloud workflow.
Now I changed as you mentioned using 'validate_each' and 
- In case of Delete action disabled.
       check eq(val(item, "storage_encrypted"), "false")
- In case of Delete action enabled.
       check logic_or(
          eq(val(item, "delete_protection"), "YES"),
         ne(val(item, "db_instance_status"), "available")
       )
Doing so, I think I have handled both the scenarios now.